### ISSUE TITLE
Public API analyzer ignores compiler-generated methods

### ISSUE THREAD

**Issue by bash** [dotnet/roslyn-analyzers#5826]
### Analyzer

**Diagnostic ID**: [RS0016](https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md)

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.PublicApiAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.PublicApiAnalyzers/3.3.3)

**Version**: 3.3.3 (Latest)

### Describe the bug
Methods generated by the compiler for `record`s generate no warning when not listed in PublicAPI.(Un)shipped.txt. Adding the members by hand causes the analyzer to generate a RS0017 warning. 

### Steps To Reproduce
1. Define a `record` (with or without a primary constructor):
   ```cs
    public record Person(string Name);
    ```
2. Apply the code fix for all RS0016 warnings.

### Expected behavior
PublicAPI.Unshipped.txt should include public compiler-generated methods,
such as `ToString()`, `Equals()`, `Deconstruct()`, `==`, etc.

### Actual behavior
PublicAPI.Unshipped.txt only contains the type, constructor and property accessors.

**Comment by bitbonk** [comment]
Alternatively (preferably?) `PublicAPI.(Un)shipped.txt` should denote that `Person` is a record.
